biz.aQute.bndlib/src/aQute/bnd/build/Workspace.java

```diff
@@ -292,24 +292,53 @@ public Workspace(File workspaceDir) throws Exception {
 	}
 
 	public Workspace(File workspaceDir, String bndDir) throws Exception {
-		super(getDefaults());
+		super(new Processor(getDefaults()));
 		this.maven = new Maven(Processor.getExecutor(), this);
 		this.layout = WorkspaceLayout.BND;
 		workspaceDir = workspaceDir.getAbsoluteFile();
 		setBase(workspaceDir); // setBase before call to setFileSystem
 		addBasicPlugin(new LoggingProgressPlugin());
 		setFileSystem(workspaceDir, bndDir);
+
+		// we must process version defaults after the
+		// normal properties are read
+
+		fixupVersionDefaults(this.getParent()
+			.getProperties());
+
 		projects = new ProjectTracker(this);
 	}
 
 	private Workspace(WorkspaceLayout layout) throws Exception {
-		super(getDefaults());
+		super(new Processor(getDefaults()));
```

Comment:
The change in this constructor does not do anything useful since there is no user-managed `cnf` folder in which they can set `-versiondefaults` to some meaningful value. This includes things using standalone workspace (bndruns, maven, gradle). They can only control a bndrun file which is read after the version defaults are already set in the standalone workspace. So this proposal only works for Bnd workspace projects but it does not work for their bndruns using `-standalone`.

Reply:
You are right, this need to be handled in the `createStandaloneWorkspace()` method after the properties are read. I've added this and some test cases for bnd, default, and standalone workspaces. 

Since these are the  only way you can create a workspace I assume maven and gradle are covered?